### ISSUE TITLE
IMAP folder per send–receive address pair

### ISSUE THREAD

**Issue by JRHaigh** [deltachat/deltachat-core#19]
Pretty much the only reason that I prefer the chat workflow to the email workflow is that chat is organised in such a way as to reduce context-switching, which is a huge hindrance to me, whereas the conventional email layout is totally chaotic – why on Earth would I want to separate a message from its replies?! The idea of separating messages by whether they were received, drafted, or sent is a ridiculous way to arrange the communications because you have all the distraction of all those irrelevant emails that you're not focused on, yet have to keep flicking back-and-to between Inbox, Draft, and Sent for _relevant_ ones (And having to check a compulsory Spam folder as well makes it that bit more annoying!), whereas sorting messages by conversation keeps more relevant messages together.
&nbsp;&nbsp;&nbsp;&nbsp;By organising messages as folder-per-conversation in the IMAP tree, they can appear as chats in Delta Chat and as folders in a conventional email client. The folder structure shouldn't assume that Delta Chat is the only email client that'll use it, so calling the folder ‘Chats’ may be a bit shortsighted. Perhaps ‘Conversations’ or ‘Pairings’, or just ‘C’. It should be user-specifiable, and be able to specify the root of the IMAP tree as the location of the conversation folders.
&nbsp;&nbsp;&nbsp;&nbsp;Conversations can be initially defined as messages matching a [send–receive address pair](https://github.com/r10s/deltachat-core/issues/13 ), such that new messages are filtered into the conversation folders by their address headers. Once this is done, however, the user would be able to move messages between conversation folders just like any other folder. This is useful for when someone changes their email address.
&nbsp;&nbsp;&nbsp;&nbsp;E.g., a folder for a send–receive address pair of JRHaigh+DeltaChat.core.GitHub.i#19@FSFE.org with alice@example.org would match emails with the following headers:

    To: Alice <alice@example.org>
    From: J. R. Haigh <JRHaigh+DeltaChat.core.GitHub.i#19@FSFE.org>
----
    To: J. R. Haigh <JRHaigh+DeltaChat.core.GitHub.i#19@FSFE.org>
    From: alice@example.net
    Reply-To: Alice <alice@example.org>

**Comment by JRHaigh** [comment]
If the send–receive address pair could be associated with the folder then it would avoid additional configuration on multiple instances of Delta Chat.
&nbsp;&nbsp;&nbsp;&nbsp;There is an IMAP extension defined in [RFC 5464](https://tools.ietf.org/html/rfc5464) that allows folders to be associated with metadata, and this could be used to store the send–receive address pair. However, it's a proposed standard and no doubt does not have ubiquitous support. Without ubiquitous support, some other solution needs to be sought anyway, for servers that do not support RFC 5464. So it seems to me that, for now, RFC 5464 is little help.
&nbsp;&nbsp;&nbsp;&nbsp;The only obvious solution to associate a folder with an address pair is to embed the address pair within the path. Furthermore, organising the send addresses as subfolders of the receive addresses – like so:

    /C/<receive address>/<send address>

– allows people who do not use multiple receive addresses to ignore that level of the folder hierarchy.
&nbsp;&nbsp;&nbsp;&nbsp;Here are some example folders:

    (Alice has 2 email addresses.)
    /C/alice@example.org/Bob@Example.org
    /C/alice@example.org/carol+axyz.w@example.net
    /C/alice@example.net/carol+axyz.h@example.net
    /C/alice@example.net/dave@example.com
    …
----
    (Bob has 1 email address.)
    /C/Bob@Example.org/alice@example.org
    /C/Bob@Example.org/carol+bxyz@example.net
    /C/Bob@Example.org/dave@example.com
    …
----
    (Carol uses subaddressing.)
    /C/carol+axyz.h@example.net/alice@example.net
    /C/carol+axyz.w@example.net/alice@example.org
    /C/carol+bxyz@example.net/Bob@Example.org
    /C/carol+dxyz@example.net/dave@example.com
    …

&nbsp;&nbsp;&nbsp;&nbsp;Note that Alice and Carol have 2 channels with each other, to separate ‘work’ and ‘home’ conversation.

**Comment by JRHaigh** [comment]
If a server _does_ support RFC 5464 then perhaps it could be used to name the channel, e.g. ‘Alice (work)’, but actually, as long as the send–receive address pair is denoted, the display name can be deduced from the address book, or one of the emails within the folder with a header value of the form ‘Alice (work) <alice@example.org>’.

**Comment by r10s** [comment]
The name of the folder is discussed at https://github.com/deltachat/deltachat-core/issues/47 ; a more complex folder structure is out of scope of Delta Chat for now, I think this may be soved using server-side filters or so.
